### PR TITLE
Adding testing for model storage and support in Github Actions

### LINKED ISSUE #116: [Feature Request] [Docs] Add Support Matrix for Models in ServerlessLLM
### Prerequisites

- [x] I have searched existing issues and reviewed documentation.

### Problem Description

To enhance transparency and user experience, ServerlessLLM will introduce a support matrix that lists supported models and their compatibility with different backends (e.g., vLLM), similar to [vLLM’s approach](https://docs.vllm.ai/en/latest/models/supported_models.html).

### Proposed Solution

The table will include:

- Model architecture (e.g., GPT, OPT, LLaMA)
- Specific models (e.g., GPT-2, OPT-1.3B, Llama-3.2-1B)
- Example HuggingFace models with links to repositories
- Backend 1 (e.g., vLLM) with a 'check' mark if supported
- Backend 2, Backend 3, etc., with 'check' marks as new backends are supported

This feature will be implemented in the following steps:

1. [ ] **Update Documentation**: Add the support matrix table to the documentation, listing supported model architectures, models, and backends.
2. [ ] **Create Testing Infrastructure**: Upload a standard test script, a file that tracks a list of supported models, and a corresponding GitHub workflow to automatically verify the models in the list.
3. [ ] **Model Testing and Validation**: Allow contributors to test `sllm` and `sllm-store` with models not in the list. When a model passes the tests, it can be added to the supported model list, and the GitHub workflow will automatically verify it.
4. [ ] **Bug Fixes**: Identify and fix any issues that arise during the testing in Step 3 to ensure robust support for all tested models.

### Alternatives Considered

_No response_

### Additional Context

_No response_

### Importance

Nice to have

### Usage Statistics (Optional)

_No response_

### PR DESCRIPTION
**Pull request by dalongbao**
## Description
Github Action that will start the sllm server via `entrypoint.sh` attempt to store every model in the list of supported models, then run inference on them. Prints which models fail the tests.

Future improvements:
* Model batching (in case storage runs out) 
* Improve how `supported_models.json` is updated

## Motivation
`close #116`

## Type of Change
- [ ] Bug fix
- [X] New feature
- [ ] Breaking change
- [X] Documentation update

## Checklist
- [X] I have read the [CONTRIBUTION](https://github.com/ServerlessLLM/ServerlessLLM/blob/main/CONTRIBUTING.md) guide.
- [X] I have updated the tests (if applicable).
- [X] I have updated the documentation (if applicable).

### CODE REVIEW

**Review by future-xy — CHANGES_REQUESTED** [review]
Thank you for your hard work on this PR! This is a fantastic start, and the quality is impressive—it’s clear you’ve put a lot of effort into it. I’ve added a few comments with best practice suggestions to make it even stronger. Please feel free to reach out here if anything is unclear—I’m here to help!

**Tips**: There are a couple of ways to test GitHub Actions locally:

1. You can emulate the steps by starting a Docker environment with the same image specified in your YAML file, then manually replay each step.
2. Alternatively, you can use a tool called [`act`](https://github.com/nektos/act), which enables you to run entire workflows locally.

Thanks again, and let me know if you have any questions!

**Comment by dalongbao** [comment]
Addressed all the comments

**Comment by future-xy** [comment]
Thank you for the updates! 

A few points to address before merging:

1. **Testing the Workflow**: Have you run the workflow locally using `act`? It seems there are still some issues to resolve:
   - There’s a format error on the first line of `inference_test.yml` (see screenshot).
   - The checkpoint store doesn’t seem to launch within the workflow, which could lead to issues in testing.
   - The `sllm` workers may also need verification, as they don’t appear to be properly initialized.

image

2. **Naming and Formatting**: Could you rename the PR to better reflect the changes and run `pre-commit` to format your code? You can refer to our [contributing guide](https://github.com/ServerlessLLM/ServerlessLLM/blob/main/CONTRIBUTING.md) for more guidance.

Let me know if you'd like any help using `act` for local testing—I can also add instructions in the contributing guide if helpful! Thanks again for your work on this.

**Comment by future-xy** [comment]
BTW, the VSCode extension "GitHub Actions" can also help with editing and validating workflow files.

File: .github/workflows/inference_test.yml
Comment (future-xy) [inline .github/workflows/inference_test.yml:53]: In the previous step, when starting the entrypoint in "HEAD" mode, the `sllm-store-server` is not started, so this test should not have a chance to pass.

File: .github/workflows/inference_test.yml
Comment (future-xy) [inline .github/workflows/inference_test.yml:36]: I would expect the two commands above to install dependencies like ray and vllm, wouldn’t they? If they don’t, that seems like a bug.

File: docs/stable/models/supported_models.md
Comment (future-xy) [inline docs/stable/models/supported_models.md:9]: Let’s remove ONNX and TensorRT. It’s probably not a good idea to mention frameworks that are unsupported (not even partially supported) here.

**Review by future-xy — CHANGES_REQUESTED** [review]

**Review by future-xy — COMMENTED** [review]
Looks good to me. I’ll approve and merge once all tests pass.

**Review by future-xy — APPROVED** [review]

**Pull request merged**
